test/config/utility.cc

```diff
@@ -369,7 +373,113 @@ envoy::config::cluster::v3::Cluster ConfigHelper::buildCluster(const std::string
     )EOF",
                                                                                  name, name,
-                                                                                 ip_version, port));
+                                                                                 address, port));
+}
+
+envoy::config::cluster::v3::Cluster ConfigHelper::buildCluster(const std::string& name) {
+  return TestUtility::parseYaml<envoy::config::cluster::v3::Cluster>(fmt::format(R"EOF(
+      name: {}
+      connect_timeout: 5s
+      type: EDS
+      eds_cluster_config: {{ eds_config: {{ ads: {{}} }} }}
+      lb_policy: ROUND_ROBIN
+      http2_protocol_options: {{}}
+    )EOF",
+                                                                                 name));
+}
+
+envoy::config::cluster::v3::Cluster ConfigHelper::buildRedisCluster(const std::string& name) {
+  return TestUtility::parseYaml<envoy::config::cluster::v3::Cluster>(fmt::format(R"EOF(
+      name: {}
+      connect_timeout: 5s
+      type: EDS
+      eds_cluster_config: {{ eds_config: {{ ads: {{}} }} }}
+      lb_policy: MAGLEV
+    )EOF",
+                                                                                 name));
+}
+
+envoy::config::endpoint::v3::ClusterLoadAssignment
+ConfigHelper::buildClusterLoadAssignment(const std::string& name, const std::string& address,
+                                         uint32_t port) {
+  return TestUtility::parseYaml<envoy::config::endpoint::v3::ClusterLoadAssignment>(
+      fmt::format(R"EOF(
+      cluster_name: {}
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: {}
+                port_value: {}
+    )EOF",
+                  name, address, port));
+}
+
+envoy::config::listener::v3::Listener ConfigHelper::buildListener(const std::string& name,
+                                                                  const std::string& route_config,
+                                                                  const std::string& address,
+                                                                  const std::string& stat_prefix) {
+  return TestUtility::parseYaml<envoy::config::listener::v3::Listener>(fmt::format(
+      R"EOF(
+      name: {}
+      address:
+        socket_address:
+          address: {}
+          port_value: 0
+      filter_chains:
+        filters:
+        - name: http
+          typed_config:
+            "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+            stat_prefix: {}
+            codec_type: HTTP2
+            rds:
+              route_config_name: {}
+              config_source: {{ ads: {{}} }}
+            http_filters: [{{ name: envoy.filters.http.router }}]
+    )EOF",
+      name, address, stat_prefix, route_config));
+}
+
+envoy::config::listener::v3::Listener ConfigHelper::buildRedisListener(const std::string& name,
```

Comment:
I might also suggest that just the generic portion (without the filters), eg `buildBaseListener` is held here, and `buildListener` appends the HCM filter to the config -- `buildRedisListener` probably won't have much use outside of ads_integration.cc, where it can append to buildBaseListener maybe?